records/generator.go

```diff
@@ -248,12 +268,38 @@ func (rg *RecordGenerator) InsertState(sj StateJSON, domain string, ns string,
 	for _, f := range sj.Frameworks {
 		fname := labels.AsDomainFrag(f.Name, spec)
 		for _, task := range f.Tasks {
-			host, ok := rg.Slaves[task.SlaveId]
+			hostIP, ok := rg.SlaveIPs[task.SlaveId]
 			// skip not running or not discoverable tasks
 			if !ok || (task.State != "TASK_RUNNING") {
 				continue
 			}
 
+			// parse docker inspect output for container IP if enabled
+			containerIP := ""
+			if useDockerContainerIP {
+				for _, status := range task.Statuses {
+					if status.State == "TASK_RUNNING" && len(status.Data) > 0 {
+						decoded, err := base64.StdEncoding.DecodeString(status.Data)
+						if err != nil {
+							logging.VeryVerbose.Printf("no base64 encoded docker-inspect output in status of task %v: ", task.Id, err)
+							continue
+						}
+
+						inspectResult := []DockerInspect{}
+						err = json.Unmarshal(decoded, &inspectResult)
+						if err != nil {
+							logging.VeryVerbose.Printf("error parsing docker inspect output of task %v: %v", task.Id, err)
+							continue
+						}
+
+						if len(inspectResult) > 0 {
+							containerIP = inspectResult[0].NetworkSettings.IPAddress
+						}
+						break
```

Comment:
I would think that you'd only want to break if the containerIP was actually found. Here we're breaking anyways.